### PR TITLE
INTYGFV-14815: Fixed redundant code that broke spotbugs. Added tests.

### PR DESCRIPTION
**Pull request by IamFriberg**
Fixed redundant code that broke spotbugs. Added tests.

### CODE REVIEW

File: support/src/main/java/se/inera/intyg/common/support/validate/ValidatorUtil.java [hunk 1]
```diff
@@ -100,39 +100,20 @@ public static boolean validateDate(InternalDate date, List<ValidationMessage> va
 
         if (!date.isValidDate()) {
             if (date.isCorrectFormat()) {
-                if (questionId != null) {
```
Comment (ChrilleSigerhed) [inline support/src/main/java/se/inera/intyg/common/support/validate/ValidatorUtil.java:103]: Can questionId not be null here?

File: support/src/main/java/se/inera/intyg/common/support/validate/ValidatorUtil.java [hunk 2]
```diff
@@ -100,39 +100,20 @@ public static boolean validateDate(InternalDate date, List<ValidationMessage> va
...
-                    addValidationErrorWithQuestionId(validationMessages, category, field, ValidationMessageType.INVALID_FORMAT,
-                        "common.validation.date_invalid", questionId);
-                } else {
-                    addValidationErrorWithQuestionId(validationMessages, category, field, ValidationMessageType.INVALID_FORMAT,
-                        "common.validation.date_invalid", questionId);
-                }
+                addValidationErrorWithQuestionId(validationMessages, category, field, ValidationMessageType.INVALID_FORMAT,
+                    "common.validation.date_invalid", questionId);
             } else if (message != null) {
```
Comment (ChrilleSigerhed) [inline support/src/main/java/se/inera/intyg/common/support/validate/ValidatorUtil.java:105]: Can message not be null here?

File: support/src/test/java/se/inera/intyg/common/support/facade/util/ValueToolkitTest.java [hunk 3]
```diff
@@ -0,0 +1,727 @@
...
+import se.inera.intyg.common.support.facade.model.value.CertificateDataValueDateList;
+import se.inera.intyg.common.support.facade.model.value.CertificateDataValueDateRange;
+import se.inera.intyg.common.support.facade.model.value.CertificateDataValueDateRangeList;
+import se.inera.intyg.common.support.facade.model.value.CertificateDataValueDiagnosis;
+import se.inera.intyg.common.support.facade.model.value.CertificateDataValueDiagnosisList;
+import se.inera.intyg.common.support.model.common.internal.GrundData;
+import se.inera.intyg.common.support.model.common.internal.HoSPersonal;
+import se.inera.intyg.common.support.model.common.internal.Vardenhet;
+
+@RunWith(MockitoJUnitRunner.class)
```
Comment (ChrilleSigerhed) [inline support/src/test/java/se/inera/intyg/common/support/facade/util/ValueToolkitTest.java:59]: Is this not JUnit 4 Annotation? @ExtendWith(MockitoExtension.class) is used in JUnit 5?

File: support/src/test/java/se/inera/intyg/common/support/facade/util/ValueToolkitTest.java [hunk 4]
```diff
@@ -0,0 +1,727 @@
...
+
+    private Map<String, CertificateDataElement> data;
+    private CertificateDataElement certificateDataElement1;
+    private CertificateDataElement certificateDataElement2;
+
+    @BeforeEach
+    void setUp() {
+        data = new HashMap<>();
+    }
+    @Nested
```
Comment (ChrilleSigerhed) [inline support/src/test/java/se/inera/intyg/common/support/facade/util/ValueToolkitTest.java:76]: Do we need to add tests with questionId = NULL & message = NULL?

File: support/src/test/java/se/inera/intyg/common/support/facade/util/ValueToolkitTest.java [hunk 5]
```diff
@@ -0,0 +1,727 @@
...
+            data.put(QUESTION_ID_1, certificateDataElement1);
+            data.put(QUESTION_ID_2, certificateDataElement2);
+
+            var response = ValueToolkit.booleanValue(data, QUESTION_ID_1, VALUE_ID_1);
+            assertNull(response);
+        }
+    }
+
+    @Nested
+    class text {
```
Comment (ChrilleSigerhed) [inline support/src/test/java/se/inera/intyg/common/support/facade/util/ValueToolkitTest.java]: Do we need to add tests with questionId = NULL & message = NULL?

File: support/src/test/java/se/inera/intyg/common/support/facade/util/ValueToolkitTest.java [hunk 6]
```diff
@@ -0,0 +1,727 @@
...
+            data.put(QUESTION_ID_1, certificateDataElement1);
+            data.put(QUESTION_ID_2, certificateDataElement2);
+
+            var response = ValueToolkit.textValue(data, QUESTION_ID_1, VALUE_ID_1);
+            assertNull(response);
+        }
+    }
+
+    @Nested
+    class uncertainDate {
```
Comment (ChrilleSigerhed) [inline support/src/test/java/se/inera/intyg/common/support/facade/util/ValueToolkitTest.java]: Do we need to add tests with questionId = NULL & message = NULL?

File: support/src/test/java/se/inera/intyg/common/support/facade/util/ValueToolkitTest.java [hunk 7]
```diff
@@ -0,0 +1,727 @@
...
+            data.put(QUESTION_ID_1, certificateDataElement1);
+            data.put(QUESTION_ID_2, certificateDataElement2);
+
+            var response = ValueToolkit.uncertainDateValue(data, QUESTION_ID_1, VALUE_ID_1);
+            assertNull(response);
+        }
+    }
+
+    @Nested
+    class icfText {
```
Comment (ChrilleSigerhed) [inline support/src/test/java/se/inera/intyg/common/support/facade/util/ValueToolkitTest.java]: Do we need to add tests with questionId = NULL & message = NULL?

File: support/src/test/java/se/inera/intyg/common/support/facade/util/ValueToolkitTest.java [hunk 8]
```diff
@@ -0,0 +1,727 @@
...
+            data.put(QUESTION_ID_1, certificateDataElement1);
+            data.put(QUESTION_ID_2, certificateDataElement2);
+
+            var response = ValueToolkit.icfTextValue(data, QUESTION_ID_1, VALUE_ID_1);
+            assertNull(response);
+        }
+    }
+
+    @Nested
+    class icfCode {
```
Comment (ChrilleSigerhed) [inline support/src/test/java/se/inera/intyg/common/support/facade/util/ValueToolkitTest.java]: Do we need to add tests with questionId = NULL & message = NULL?

File: support/src/test/java/se/inera/intyg/common/support/facade/util/ValueToolkitTest.java [hunk 9]
```diff
@@ -0,0 +1,727 @@
...
+            data.put(QUESTION_ID_1, certificateDataElement1);
+            data.put(QUESTION_ID_2, certificateDataElement2);
+
+            var response = ValueToolkit.icfCodeValue(data, QUESTION_ID_1, VALUE_ID_1);
+            assertNull(response);
+        }
+    }
+
+    @Nested
+    class dateList {
```
Comment (ChrilleSigerhed) [inline support/src/test/java/se/inera/intyg/common/support/facade/util/ValueToolkitTest.java]: Do we need to add tests with questionId = NULL & message = NULL?

File: support/src/test/java/se/inera/intyg/common/support/facade/util/ValueToolkitTest.java [hunk 10]
```diff
@@ -0,0 +1,727 @@
...
+            data.put(QUESTION_ID_1, certificateDataElement1);
+            data.put(QUESTION_ID_2, certificateDataElement2);
+
+            var response = ValueToolkit.dateListValue(data, QUESTION_ID_1, VALUE_ID_1);
+            assertNull(response);
+        }
+    }
+
+    @Nested
+    class date {
```
Comment (ChrilleSigerhed) [inline support/src/test/java/se/inera/intyg/common/support/facade/util/ValueToolkitTest.java]: Do we need to add tests with questionId = NULL & message = NULL?

File: support/src/test/java/se/inera/intyg/common/support/facade/util/ValueToolkitTest.java [hunk 11]
```diff
@@ -0,0 +1,727 @@
...
+            data.put(QUESTION_ID_2, certificateDataElement2);
+
+            var response = ValueToolkit.dateValue(data, QUESTION_ID_1, VALUE_ID_1);
+            assertNull(response);
+        }
+
+    }
+
+    @Nested
+    class code {
```
Comment (ChrilleSigerhed) [inline support/src/test/java/se/inera/intyg/common/support/facade/util/ValueToolkitTest.java]: Do we need to add tests with questionId = NULL & message = NULL?

File: support/src/test/java/se/inera/intyg/common/support/facade/util/ValueToolkitTest.java [hunk 12]
```diff
@@ -0,0 +1,727 @@
...
+            data.put(QUESTION_ID_1, certificateDataElement1);
+            data.put(QUESTION_ID_2, certificateDataElement2);
+
+            var response = ValueToolkit.codeValue(data, QUESTION_ID_1);
+            assertNull(response);
+        }
+    }
+
+    @Nested
+    class codeList {
```
Comment (ChrilleSigerhed) [inline support/src/test/java/se/inera/intyg/common/support/facade/util/ValueToolkitTest.java]: Do we need to add tests with questionId = NULL & message = NULL?

File: support/src/test/java/se/inera/intyg/common/support/facade/util/ValueToolkitTest.java [hunk 13]
```diff
@@ -0,0 +1,727 @@
...
+            data.put(QUESTION_ID_1, certificateDataElement1);
+            data.put(QUESTION_ID_2, certificateDataElement2);
+
+            var response = ValueToolkit.codeListValue(data, QUESTION_ID_1);
+            assertTrue(response.isEmpty());
+        }
+    }
+
+    @Nested
+    class diagnosisList {
```
Comment (ChrilleSigerhed) [inline support/src/test/java/se/inera/intyg/common/support/facade/util/ValueToolkitTest.java]: Do we need to add tests with questionId = NULL & message = NULL?

File: support/src/test/java/se/inera/intyg/common/support/facade/util/ValueToolkitTest.java [hunk 14]
```diff
@@ -0,0 +1,727 @@
...
+            data.put(QUESTION_ID_1, certificateDataElement1);
+            data.put(QUESTION_ID_2, certificateDataElement2);
+
+            var response = ValueToolkit.diagnosisListValue(data, QUESTION_ID_1);
+            assertTrue(response.isEmpty());
+        }
+    }
+
+    @Nested
+    class dateRangeList {
```
Comment (ChrilleSigerhed) [inline support/src/test/java/se/inera/intyg/common/support/facade/util/ValueToolkitTest.java]: Do we need to add tests with questionId = NULL & message = NULL?

File: support/src/test/java/se/inera/intyg/common/support/facade/util/ValueToolkitTest.java [hunk 15]
```diff
@@ -0,0 +1,727 @@
...
+            data.put(QUESTION_ID_1, certificateDataElement1);
+            data.put(QUESTION_ID_2, certificateDataElement2);
+
+            var response = ValueToolkit.dateRangeListValue(data, QUESTION_ID_1);
+            assertTrue(response.isEmpty());
+        }
+    }
+
+    @Nested
+    class grund {
```
Comment (ChrilleSigerhed) [inline support/src/test/java/se/inera/intyg/common/support/facade/util/ValueToolkitTest.java]: Do we need to add tests with questionId = NULL & message = NULL?

Comment (ChrilleSigerhed) [inline support/src/test/java/se/inera/intyg/common/support/facade/util/ValueToolkitTest.java]: Should this class be called GrundData?

**Review by ChrilleSigerhed — CHANGES_REQUESTED** [review]
Great job, good that you added tests for ValueToolkit!

File: support/src/main/java/se/inera/intyg/common/support/facade/util/ValueToolkit.java [hunk 16]
```diff
@@ -144,7 +144,7 @@ public static LocalDate dateValue(Map<String, CertificateDataElement> data, Stri
         }
 
         final var dateValue = (CertificateDataValueDate) dataValue;
-        if (dateValue == null || dateValue.getId() == null || !dateValue.getId().equalsIgnoreCase(valueId)) {
+        if (dateValue.getId() == null || !dateValue.getId().equalsIgnoreCase(valueId)) {
```
[the remark this answers is not in the document]
Comment (IamFriberg) [reply support/src/main/java/se/inera/intyg/common/support/facade/util/ValueToolkit.java:147]: It has already been checked above when we do the instance of check.

**Review by ChrilleSigerhed — APPROVED** [review]
